generator/feature_sorter.cpp

```diff
@@ -155,17 +154,27 @@ class FeaturesCollector2 : public FeaturesCollector
     {
     bool const isLine = fb.IsLine();
     bool const isArea = fb.IsArea();
-    CHECK(!isLine || !isArea, ("A feature can't have both points and triangles geometries:", fb.GetMostGenericOsmId()));
+    CHECK(!isLine || !isArea, (fb.GetMostGenericOsmId()));
 
     int const scalesStart = static_cast<int>(m_header.GetScalesCount()) - 1;
     for (int i = scalesStart; i >= 0; --i)
     {
-      int const level = m_header.GetScale(i);
+      int level = m_header.GetScale(i);
       // TODO : re-checks geom limit rect size via IsDrawableForIndexGeometryOnly() which was checked already in CalculateMidPoints.
       if (fb.IsDrawableInRange(scales::PatchMinDrawableScale(i > 0 ? m_header.GetScale(i - 1) + 1 : 0),
                                scales::PatchMaxDrawableScale(level)))
       {
         bool const isCoast = fb.IsCoastCell();
+        // Increment zoom level for coastline polygons (check and simplification)
+        // for better visual quality in the first geometry batch or whole WorldCoasts.
+        if (isCoast)
+        {
+          if (level <= scales::GetUpperWorldScale())
+            ++level;
+          if (i == 0)
+            ++level;
```

Comment:
won't it be better to make coasts' trg0 fallback to trg1 always?
or explicitly keep only 3 geom levels for coasts (expand trg0 to `[1-5]` instead of `[1-3]`)

Reply:
maybe just leave a todo comment for now if you agree

Reply:
Yes, trg0 -> 1 fallback should do the trick.

Reply:
so it makes trg0 coasts in country mwms unnecessary as they duplicate trg3 in WorldCoasts :) might save a few kilobytes someday in the future :)